src/gui/properties/speedplotview.cpp

```diff
@@ -33,15 +33,60 @@
 #include "base/global.h"
 #include "base/utils/misc.h"
 
+namespace
+{
+    enum PeriodInSeconds
+    {
+        MIN1_SEC = 60,
+        MIN5_SEC = 5 * 60,
+        MIN30_SEC = 30 * 60,
+        HOUR6_SEC = 6 * 60 * 60
+    };
+
+    const int MIN5_BUF_SIZE = 5 * 60;
+    const int MIN30_BUF_SIZE = 5 * 60;
+    const int HOUR6_BUF_SIZE = 5 * 60;
+    const int divider30Min = int(MIN30_SEC) / int(MIN30_BUF_SIZE);
+    const int divider6Hour = int(HOUR6_SEC) / int(HOUR6_BUF_SIZE);
+};
+
+SpeedPlotView::Averager::Averager(int divider, boost::circular_buffer<PointData> &sink)
+    : m_counter(0)
+    , m_divider(divider)
+    , m_sink(sink)
+    , m_accumulator({})
+{
+}
+
+inline void SpeedPlotView::Averager::push(const PointData &p)
+{
+    m_accumulator.x += p.x;
+    for (int id = UP; id < NB_GRAPHS; ++id)
+        m_accumulator.y[id] += p.y[id];
+    m_counter = (m_counter + 1) % m_divider;
+    if (m_counter == 0) {
+        for (int id = UP; id < NB_GRAPHS; ++id)
+            m_accumulator.y[id] /= m_divider;
+        m_accumulator.x /= m_divider;
+        // now flush out averaged data
+        m_sink.push_back(m_accumulator);
+        m_accumulator = {};
+    }
+}
+inline bool SpeedPlotView::Averager::isReady()
```

Comment:
Don't forget to split method definitions with blank line.

Reply:
Please don't use `inline` at all. We prefer to trust the compiler to decide what should be inlined.